Modules/StudyProgramme/classes/class.ilObjStudyProgrammeListGUI.php

```diff
@@ -68,6 +75,90 @@ public function getCommandLink(string $cmd): string
         return $this->ctrl->getLinkTargetByClass("ilobjstudyprogrammegui", $cmd);
     }
 
+    public function getAsListItem(int $ref_id, int $obj_id, string $type, string $title, string $description): ?\ILIAS\UI\Component\Item\Item
+    {
+        $this->initItem(
+            $ref_id,
+            $obj_id,
+            $type,
+            $title,
+            $description
+        );
+        $this->insertCommands(true, true);
+
+        $prg = new ilObjStudyProgramme($ref_id);
+        $assignments = $prg->getAssignments();
+        if ($this->getCheckboxStatus() && count($assignments) > 0) {
+            $this->setAdditionalInformation($this->lng->txt("prg_can_not_manage_in_repo"));
+            $this->enableCheckbox(false);
+        } else {
+            $this->setAdditionalInformation(null);
+        }
+        /** @var ilStudyProgrammeUserTable $user_table */
+        $user_table = ilStudyProgrammeDIC::dic()['ilStudyProgrammeUserTable'];
+        $user_table->disablePermissionCheck(true);
+        $data = $user_table->fetchData($prg->getId(), [$this->user->getId()]);
+        $properties = [];
+        if (count($data) === 1) {
+            $data = $data[0];
+
+            $min = $data->getPointsRequired();
+            $max = $data->getPointsReachable();
+            $cur = $data->getPointsCurrent();
+            $required_string = $min;
+            if ((float) $max < (float) $min) {
+                $required_string .= ' ' . $this->lng->txt('prg_dash_label_unreachable') . ' (' . $max . ')';
+            }
+
+            $properties = [
+                [$this->lng->txt('prg_dash_label_minimum') => $required_string],
+                [$this->lng->txt('prg_dash_label_gain') => $cur],
+                [$this->lng->txt('prg_dash_label_status') => $data->getStatus()],
+            ];
+
+            if (in_array(
+                $data->getStatusRaw(),
+                [ilPRGProgress::STATUS_COMPLETED, ilPRGProgress::STATUS_ACCREDITED],
+                true
+            )) {
+                $validity = $data->getExpiryDate() ?: $data->getValidity();
+                $properties[] = [$this->lng->txt('prg_dash_label_valid') => $validity];
+            } else {
+                $properties[] = [$this->lng->txt('prg_dash_label_finish_until') => $data->getDeadline()];
+            }
+
+            $validator = new ilCertificateDownloadValidator();
+            if ($validator->isCertificateDownloadable($data->getUsrId(), $data->getNodeId())) {
+                $cert_url = "ilias.php?baseClass=ilRepositoryGUI&ref_id=" . $prg->getRefId(
+                ) . "&cmd=deliverCertificate";
```

Comment:
Don't append parameters to URLs this way (I am sure you copied this code fragment from somewhere). Use `ilCtrl` for this.